src/screens/settings/__tests__/settings.test.tsx

```diff
@@ -1,12 +1,64 @@
 import 'react-native';
 import React from 'react';
-import { SettingsScreen } from '../settings';
-
+import { SettingsScreenComponent, IProps, IReduxProps } from '../settings';
 import { shallow } from 'enzyme';
+import styleProvider from '../styles';
+import { darkTheme } from '../../../styles/themes/dark-theme';
+
+const props: IProps & IReduxProps = {
+    // @ts-ignore
+    navigation: {
+        navigate: jest.fn()
+    },
+    styles: styleProvider(darkTheme),
+    theme: darkTheme,
+    mock: jest.fn()
+};
 
 describe('settings screen component', () => {
+    const wrapper = shallow(<SettingsScreenComponent {...props} />);
+
     it('renders correctly', () => {
-        const wrapper = shallow(<SettingsScreen />);
         expect(wrapper.debug()).toMatchSnapshot();
     });
+    it('Pin login switch tapped', () => {
+        wrapper.find('[testID="pin-login"]').simulate('valueChange');
+        expect(props.mock).toHaveBeenCalled();
```

Comment:
Since prop is a global value, i think that before each test we should clear mock, i think toHaveBeenCalled() will pass in all cases if one call was made.
https://jestjs.io/docs/en/mock-function-api.html#mockfnmockclear

Reply:
Corect